### PR TITLE
feat: Display test names properly with metric grouping, add support for on-the-fly specification of metrics.

### LINKED ISSUE #83: `swift package benchmark --grouping metric` missing benchmark names
In the current `main`, (March 4th - commit: fa1a95596f5892361f909f5858246e73c2e476dc) - when you use the option `--grouping metric`, the benchmark names are removed.

Examples:
 - the stock `swift package benchmark` [command output](https://gist.github.com/heckj/62f4be896f65ef165c41b2712048416e)
 - `swift package benchmark --grouping metric` [command output](https://gist.github.com/heckj/a49e18bde9a7676b8c699c28b043f831)

### LINKED ISSUE #95: feat: Add support for specifying exact metrics to run dynamically
E.g. support a `--metric` command line option.

### PR DESCRIPTION
**Pull request by hassila**
## Description

Fixes #83 
Fixes #95

## How Has This Been Tested?

Manual testing.

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [x] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hassila** [comment]
Current dependencies on/for this PR:
* main
  * **PR #96** <a href="https://app.graphite.dev/github/pr/ordo-one/package-benchmark/96" target="_blank">Graphite</a> 
    * **PR #97** <a href="https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97?utm_source=stack-comment).

File: Sources/Benchmark/BenchmarkMetric.swift [hunk 1]
```diff
@@ -186,9 +200,9 @@ public extension BenchmarkMetric {
 #if swift(>=5.8)
     @_documentation(visibility: internal)
 #endif
-extension BenchmarkMetric {
-    init(_ textualMetric: String) {
-        switch textualMetric {
+public extension BenchmarkMetric {
+    init?(argument: String) {
```
Comment (mr-swifter) [inline Sources/Benchmark/BenchmarkMetric.swift:204]: Why fallible? It seems object will be always created?

**Review by mr-swifter — APPROVED** [review]

File: Sources/Benchmark/BenchmarkMetric.swift (hunk 1 above)
Comment (hassila) [reply Sources/Benchmark/BenchmarkMetric.swift:204]: It is to conform to ExpressibleByArgument protocol from SAP.

Comment (hassila) [reply Sources/Benchmark/BenchmarkMetric.swift:204]: https://apple.github.io/swift-argument-parser/documentation/argumentparser/expressiblebyargument/

**Review by dimlio — APPROVED** [review]

**Comment by hassila** [comment]
@hassila queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97).

**Comment by hassila** [comment]
[Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97) couldn't merge this pull request because a downstack PR #96 failed to merge.

**Comment by hassila** [comment]
@hassila queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97).

**Comment by hassila** [comment]
[Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97) rebased this pull request as part of a merge.

**Comment by hassila** [comment]
@hassila merged this pull request with [Graphite](https://app.graphite.dev/github/pr/ordo-one/package-benchmark/97).

**Pull request merged**